pkgs/os-specific/linux/device-tree/default.nix

```
@@ -1,16 +1,30 @@
 { stdenvNoCC, dtc, findutils }:
 
 with stdenvNoCC.lib; {
-  applyOverlays = (base: overlays: stdenvNoCC.mkDerivation {
+  applyOverlays = (base: overlays': stdenvNoCC.mkDerivation {
     name = "device-tree-overlays";
     nativeBuildInputs = [ dtc findutils ];
     buildCommand = let
-      quotedDtbos = concatMapStringsSep " " (o: "\"${toString o}\"") (toList overlays);
+      overlays = toList overlays';
     in ''
-      for dtb in $(find ${base} -name "*.dtb" ); do
-        outDtb=$out/$(realpath --relative-to "${base}" "$dtb")
-        mkdir -p "$(dirname "$outDtb")"
-        fdtoverlay -o "$outDtb" -i "$dtb" ${quotedDtbos};
+      mkdir -p $out
+      for dtb in $(find ${base} -name '*.dtb'); do
+        cp $dtb $out
+      done
```

It seems to me that uboot requires the same directory structure (e.g. `broadcom/*.dtb` instead of just `*.dtb`) as the kernel source. This code just copies the files into one flat directory. I propose a similar change as in `filterDTBs`:

```suggestion
      cd ${base}
      find . -type f -name '*.dtb' -print0 \
        | xargs -0 cp -v --no-preserve=mode --target-directory $out --parents
```

---

While not strictly needed it's a nice improvement, switched to your suggestion.